src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -43,6 +43,19 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 		this._toDispose = dispose(this._toDispose);
 	}
 
+	public $getConnections(activeConnectionsOnly?: boolean): Thenable<azdata.connection.ConnectionProfile[]> {
+		let connections = this._connectionManagementService.getActiveConnections();
+		if (activeConnectionsOnly !== true) {
+			this._connectionManagementService.getRecentConnections().forEach(connection => {
+				if (connections.filter(activeConnection => activeConnection.id === connection.id).length === 0) {
+					connections.push(connection);
```

Comment:
I'd do 

if(!connections.any(activeConnection => activeConnection.id === connection.id)) {
   connections.push(connection);
}

so you short-circuit in the case that the connection is in the active connection list

Reply:
any does not exist for array in js :)

Reply:
Fair, you can use find to accomplish the same thing. Not the end of the world though if you'd prefer to keep it this way.

Reply:
I think a better way is to write it this way -
```
this._connectionManagementService.getRecentConnections().forEach(connection => {
    if (!connections.includes(connection)) {
        connections.push(connection);
    }
});
```
`connections.filter` creates a new array every time since it's in a `foreach` and that's a waste of space.